parser/parser.y

```diff
@@ -2851,19 +2854,35 @@ FunctionCallConflict:
 		$$ = &ast.BinaryOperationExpr{Op: opcode.Mod, L: $3.(ast.ExprNode), R: $5.(ast.ExprNode)}
 	}
 
+DistinctKwd:
+	"DISTINCT"
+|	"DISTINCTROW"
+
 DistinctOpt:
+	"ALL"
 	{
 		$$ = false
 	}
-|	"ALL"
+|	DistinctKwd
+	{
+		$$ = true
+	}
+
+DefaultFalseDistinctOpt:
 	{
 		$$ = false
 	}
-|	"DISTINCT"
+|	DistinctOpt
+
+DefaultTrueDistinctOpt:
 	{
 		$$ = true
 	}
-|	"DISTINCT" "ALL"
```

Comment:
Ditto. `"DISTINCT" "ALL"` moved to `BuggyXxxxx` since normally those two (mutually-exclusive) keywords should not be allowed to appear together.